plugins/rest/komand_rest/actions/post/action.py

```diff
@@ -12,12 +13,19 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        response = self.connection.api.call_api(
-            method="POST",
-            path=params.get(Input.ROUTE),
-            json_data=params.get(Input.BODY, {}),
-            headers=params.get(Input.HEADERS, {}),
-        )
+        headers = params.get(Input.HEADERS, {})
+        switch_ONoff = check_headers_for_urlencoded(headers)
+
+        body = params.get(Input.BODY, {})
+        body = convert_body_for_urlencoded(headers, body)
+
+        args = {"method": "POST", "path": params.get(Input.ROUTE), "data": body, "headers": headers}
+        args2 = {"method": "POST", "path": params.get(Input.ROUTE), "json_data": body, "headers": headers}
+
+        if switch_ONoff:
+            response = self.connection.api.call_api(**args)
+        else:
+            response = self.connection.api.call_api(**args2)
 
         return {
             Output.BODY_OBJECT: Common.body_object(response),
```

Comment:
```suggestion
        headers = params.get(Input.HEADERS, {})
        body = params.get(Input.BODY, {})
        if check_headers_for_urlencoded(headers):
            body = convert_body_for_urlencoded(headers, body)
            kwargs = {"method": "POST", "path": params.get(Input.ROUTE), "data": body, "headers": headers}
        else:
            kwargs = {"method": "POST", "path": params.get(Input.ROUTE), "json_data": body, "headers": headers}
            
        response = self.connection.api.call_api(**kwargs)
        return {
            Output.BODY_OBJECT: Common.body_object(response),
```
I would do something like that. Please test it though.

Reply:
+1 😄